### ISSUE TITLE
Layout doesn't seem to change

### ISSUE THREAD

**Issue by samtsai** [abovethewater/grunt-docco-multi#1]
I've had an issue where changing the layout does not change the outputted files.

I know that it correctly looks for the layout, because in my fork I'm calling a custom layout and it errors out when the layout did not exist.

One trick to get it to work is to add the single file option for docco:
    docco: {
      all: {
        files:
          {
            src: [
              '<%= common_js %>**/_.js',
              '!<%= common_js %>\_*/min/_',
...
            ]
          }
      },
      *_single: ['Gruntfile.js']**,
      options: {
        layout: 'pretty',
        output: 'docs/'
      }
    },

Then it works. I guess what I'm finding is that I need to always have the single option.

**Comment by samtsai** [comment]
Ok I think I misunderstood the README.

"single" is just an example that you gave, but for whatever reason, including a single file triggered the layout change.

Here's my grunt docco --verbose output:

Running "docco:all" (docco) task
Verifying property docco.all exists in config...OK
Files: modules/WebCommon/j2ee-apps/webcommon.war/javascript/RadioCarousel.js, modules/WebCommon/j2ee-apps/webcommon.war/javascript/aefb.js, modules/WebCommon/j2ee-apps/webcommon.war/javascript/aehelper.js, modules/WebCommon/j2ee-apps/webcommon.war/javascript/asap.js...
...output too long...

> > 193 files doccoed.

**Comment by abovethewater** [comment]
Are you using the latest code, or the npm version?

I can't replicate this, though I have added some more test files to try to.

Can you try the following in your config:

```
all: {
    options: {
      layout: 'pretty',
      output: 'docs/'
    },
    files: {
        src: [
            '<%= common_js %>**/*.js'
        ]
    }
}
```

i.e. put the options in the task specification.

I've updated the README to try to prevent future confusion.

**Comment by abovethewater** [comment]
Ok, I've tested with the config from your fork and looks to work fine.

package.json:

```
"docco": "git+https://github.com/dianadujing/docco.git#jump_menu" 
```

Using the following config:

```
    docco:
        build:
            expand: true
            cwd: './src/'
            src: ['**/*.coffee']
            options:
                layout: 'pretty'
                output: 'docs/'
```

Produces a site with a "pretty" blue background that is not available in the default docco.

```
>> done
>> 47 files doccoed.

Done, without errors.
```

I have pushed version 0.0.2 to npm, as I have tweaked a couple of bits.

If you are still having issues, then try clearing your npm cache, as it may not be using the expected version of the git dependency.

**Comment by samtsai** [comment]
Thanks for your follow up.

I've grabbed 0.0.2 (removed node_modules and reinstalled to be sure). I've tried what you mentioned above but it's not quite working yet.

I've been also trying building your branch and while all the tests pass, the layouts don't seem to support multiple. I'd expect "docco.css" to be broken out per layout, but it seems that it's more of a mish-mash so the Gruntfile with the "linear" layout looks like this:
![screen shot 2013-07-05 at 10 03 30 am](https://f.cloud.github.com/assets/225526/753989/8fdc3320-e57b-11e2-9c2f-795bd90ea56b.png)

**Comment by abovethewater** [comment]
Not entirely sure what you are trying to achieve here. I'm sure we can make
it accommodate what you need.

Are you trying to document many files using different formats for varying
ones?

The multi is more for the fact that 1) docco already exists in npm (though
didn't work for me) 2) docco2 bails out with too many open files on OSX for
me. This "multi" version works with lots of files.

This has only been up for a week at most, so I'm happy to take your use
case and run with it.

Do you have an example repo or test case (with Gruntfile)

I'm out for the next day or so, but will look into this when I get back.

**Comment by samtsai** [comment]
Likewise I'll be out for the next week and a half but appreciate your help.
​
I'd love to share more when I get back! 

—

**Comment by samtsai** [comment]
Hi again,

The primary issue I'm trying to resolve is to understand why some filesets don't completely get built-out / doccoed. There are no warning/error messages and in some cases the process never gets to build out the docco.css and the public folder within docs.

Docco reports: >> 207 files doccoed but when I open the docs folder I see only 34 files (used: ls | wc -l)

I've tried with the master branch of your original repo as well as my fork. Same results.

**Comment by abovethewater** [comment]
Hi,

Out of interest, do all of these files have duplicate names.
Docco flattens everything, so it tends to overwrite files with the same
name.

There is an issue with docco #223 that covers this.

Also there is a fixed issue #209 that covers the CSS not being copied.

Either of these could well be causing your issue.

Try using the master docco branch instead of the npm version to see if it
helps.

Worth looking at the issue list for github.com/jashkenas/docco.

Let me know if that helps, if not I'll look further into it. (Can work
around both issues if needed)

Joe

**Comment by abovethewater** [comment]
Closing as no response in 2 years..
